diffkemp/simpll/Result.h

```diff
@@ -106,19 +120,37 @@ struct NonFunctionDifference {
 
 /// Syntactic difference between objects that cannot be found in the original
 /// source files.
-/// Note: this can be either a macro difference or inline assembly difference.
+/// There are multiple kinds of the differences: a macro diff,
+/// inline assembly diff, function-macro diff or macro-function diff.
 struct SyntaxDifference : public NonFunctionDifference {
+    enum SyntaxKind {
+        MACRO,
+        ASSEMBLY,
+        FUNCTION_MACRO,
+        MACRO_FUNCTION,
+        UNKNOWN
+    };
+    SyntaxKind syntaxKind;
     /// The difference.
     std::string BodyL, BodyR;
-    SyntaxDifference() : NonFunctionDifference(SynDiff){};
-    SyntaxDifference(std::string name,
+    // Information about definitions of differing 'object'.
+    // For now it is only used in macro differences, for other differences it is
+    // null.
+    std::unique_ptr<Definition> diffDefL, diffDefR;
+    SyntaxDifference()
+            : NonFunctionDifference(SynDiff), syntaxKind(SyntaxKind::UNKNOWN){};
+    SyntaxDifference(SyntaxKind syntaxKind,
+                     std::string name,
                      std::string BodyL,
                      std::string BodyR,
                      CallStack StackL,
                      CallStack StackR,
-                     std::string function)
+                     std::string function,
+                     std::unique_ptr<Definition> diffDefL,
+                     std::unique_ptr<Definition> diffDefR)
```

Comment:
> Any reason we need std::unique_ptr here? Why not just plain `Definition`

I used `unique_ptr`, because I wanted it to be nullptr in case where there is no saved definition eg. for assembly difference or right now also for `macro-function`/`function-macro` differences.